packages/api/src/hooks/useOnfido.ts

```diff
@@ -0,0 +1,126 @@
+import { useCallback, useRef, useMemo, useEffect, useState, useId } from 'react';
+import useOnfidoServiceToken from './useOnfidoServiceToken';
+import { ALPHA_2_TO_ALPHA_3, ONFIDO_PHRASES } from '../constants';
+import useSettings from './useSettings';
+import useResidenceList from './useResidenceList';
+import type { SdkHandle, SdkResponse } from '../types/onfido';
+import useOnfidoNotificationEvent from './useOnfidoNotificationEvent';
+import { v4 as uuidv4 } from 'uuid';
+
+/** A custom hook to initialize Onfido SDK  */
+const useOnfido = () => {
+    const [documentsIds, setDocumentIds] = useState<string[]>([]);
+    const [isOnfidoLoaded, setIsOnfidoLoaded] = useState(false);
+
+    const onfidoRef = useRef<SdkHandle | null>(null);
+    const onfidoContainerId = useMemo(() => uuidv4(), []);
+
+    const { data: settings } = useSettings();
+    const { data: residenceList } = useResidenceList();
+    const {
+        data: { token },
+    } = useOnfidoServiceToken();
+    useOnfidoNotificationEvent(documentsIds);
+
+    const countryCode = useMemo(() => {
+        return settings?.country_code || '';
+    }, [settings?.country_code]);
+
+    const supportedDocuments = useMemo(() => {
+        if (countryCode && residenceList) {
+            const onfidoResidence = residenceList.find(residence => residence?.value === countryCode).identity.services
+                ?.onfido;
+            // if we have onfido enabled for the country
+            if (onfidoResidence && onfidoResidence?.is_country_supported) {
+                return Object.keys(onfidoResidence?.documents_supported || {}).map(
+                    (document: string) => onfidoResidence?.documents_supported[document].display_name
+                );
+            }
+        }
+        return [];
+    }, [residenceList, countryCode]);
+
+    const onComplete = useCallback((data: Omit<SdkResponse, 'data'> & { data?: { id?: string } }) => {
+        const document_ids = Object.keys(data).map(key => data[key as keyof SdkResponse]?.id || '');
+        onfidoRef?.current?.safeTearDown();
+        setDocumentIds(document_ids);
+    }, []);
+
+    const loadOnfidoSdkScript = () => {
+        if (!isOnfidoLoaded) {
```

Comment:
not sure if that use case is possible, but what if I try to load onfido again, once it started, but before previous loading finished?

Reply:
Actually its a miss on my part, we shouldn't be re-loading the onfido script if it already exists in the DOM 😓 , if `useOnfido` is called in other different pages it shouldn't reload the script again if its already downloaded from CDN

Reply:
for the case where onfido is still loading and we attempt to re-load it, I'll add another check to prevent re-loading the script if onfiod has not finished loading yet

Reply:
There is a hook in usehooks-ts for this though 😮 : https://usehooks-ts.com/react-hook/use-script

Reply:
Good stuff, I really do not like that kind of edge cases. One edge case like that is not a big deal, but if we leave more of them, whole thing can became brittle.